### PR TITLE
add greenhouse

### PR DESCRIPTION
**Pull request by xianny**
## Description

Start scraping Greenhouse info.

Env variables
`GREENHOUSE_ACCESS_TOKEN`: required
`GREENHOUSE_START_DATE`: only scrape records updated after this date. If missing, will scrape records after the most recent saved record or a default fallback date.
`GREENHOUSE_END_DATE`: only scrape records updated before this date. If missing, falls back to current time.

I chose to save dates as ISO strings because they are more human readable than timestamps and still unambiguous. They may be less efficient for indexing, but this table shouldn't have a lot of data in it. Open to discussion. 

## Testing instructions

Run unit tests with `yarn test`
Run `node lib/src/scripts/pull_greenhouse.js` to try the script

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

**Comment by MaCaki** [comment]
Question (but disregard if this is impossible):  could the parameters of this script be made into cmd-line args?  I'm kind of thinking about how to make the interface between the DockerOperator and the data scripts uniform with respect to date partitions and required metadata.

**Comment by xianny** [comment]
@MaCaki , yep, totally possible. Is this something I should change before merging, or something we should think about and then change across the board? 

Just tested this on QA and everything works fine.

**Comment by askeluv** [comment]
Might be worth adding some tests in `entities` to ensure there are no datatype issues etc.

File: packages/pipeline/src/scripts/pull_greenhouse.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+function getEndDate(): Date {
+    if (process.env.GREENHOUSE_END_DATE) {
+        return new Date(process.env.GREENHOUSE_END_DATE);
+    } else {
+        return new Date();
+    }
+}
+async function fetchAndSaveApplicationsAsync(source: GreenhouseSource): Promise<void> {
+    const repository = connection.getRepository(GreenhouseApplication);
```
Comment (feuGeneA) [inline packages/pipeline/src/scripts/pull_greenhouse.ts]: Insert newline.

File: packages/pipeline/migrations/1551465374766-CreateGreenhouseTables.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { MigrationInterface, QueryRunner, Table, TableIndex } from 'typeorm';
+
+const applications = new Table({
+    name: 'raw.greenhouse_applications',
+    columns: [
+        { name: 'id', type: 'bigint', isPrimary: true },
+        { name: 'candidate_id', type: 'bigint' },
+        { name: 'applied_at', type: 'timestamp' },
+        { name: 'rejected_at', type: 'timestamp', isNullable: true },
+        { name: 'last_activity_at', type: 'timestamp', isPrimary: true },
```
[the remark this answers is not in the document]
Comment (xianny) [reply packages/pipeline/migrations/1551465374766-CreateGreenhouseTables.ts]: gotcha, I was keeping the order that they appeared in the API, but will rearrange

**Comment by xianny** [comment]
build error is because `/contracts-exchange` is automatically using a new version of Solidity (0.5.6). Fix is happening separately. This still works on QA so should be safe to merge.

**Comment by MaCaki** [comment]
@xianny sorry I just saw you reply to my comment: no I don't think it makes sense to include that change in this PR.  I'm just sort of thinking about how all these scripts might be made to fit a standard command line interface so that the DockerOperator can directly pass in DAG-task like parameters:  somthing like `run time, start_time, end_time, extra_args`.  I'm going to keep reading through pipelines though to see how it can be pulled out as a stand alone package.

**Pull request merged**